File: GeneratedEntityReaderClasses/Components/IKLimbWalkerComponent.cs
namespace GeneratedNoitaClasses;
public class IKLimbWalkerComponent : NoitaComponentBase,  iNoitaType<IKLimbWalkerComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaFloat ground_attachment_min_spread { get; set; } = new();
	public NoitaInt ground_attachment_max_tries { get; set; } = new();
	public NoitaFloat ground_attachment_max_angle { get; set; } = new();
	public NoitaFloat ground_attachment_ray_length_coeff { get; set; } = new();
	public NoitaFloat leg_velocity_coeff { get; set; } = new();
	public NoitaBool affect_flying { get; set; } = new();
	public NoitaInt ray_skip_material { get; set; } = new();
	public NoitaCVector2<NoitaFloat> mTarget { get; set; } = new();
	public NoitaCVector2<NoitaFloat> mPrevTarget { get; set; } = new();
	public NoitaCVector2<NoitaFloat> mPrevCenterPosition { get; set; } = new();
	public NoitaInt mState { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		ground_attachment_min_spread.Read(s);
		ground_attachment_max_tries.Read(s);
		ground_attachment_max_angle.Read(s);
		ground_attachment_ray_length_coeff.Read(s);
		leg_velocity_coeff.Read(s);
		affect_flying.Read(s);
		ray_skip_material.Read(s);
		mTarget.Read(s);
		mPrevTarget.Read(s);
		mPrevCenterPosition.Read(s);
		mState.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		ground_attachment_min_spread.Write(s);
		ground_attachment_max_tries.Write(s);
		ground_attachment_max_angle.Write(s);
		ground_attachment_ray_length_coeff.Write(s);
		leg_velocity_coeff.Write(s);
		affect_flying.Write(s);
		ray_skip_material.Write(s);
		mTarget.Write(s);
		mPrevTarget.Write(s);
		mPrevCenterPosition.Write(s);
		mState.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/IKLimbWalkerComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaFloat
from BasicNoitaTypes import NoitaInt
from BasicNoitaTypes import NoitaCVector2
from BasicNoitaTypes import NoitaFloat

from NoitaStream import NoitaStream
@dataclass
class IKLimbWalkerComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    ground_attachment_min_spread:NoitaFloat = field(default_factory=NoitaFloat)
    ground_attachment_max_tries:NoitaInt = field(default_factory=NoitaInt)
    ground_attachment_max_angle:NoitaFloat = field(default_factory=NoitaFloat)
    ground_attachment_ray_length_coeff:NoitaFloat = field(default_factory=NoitaFloat)
    leg_velocity_coeff:NoitaFloat = field(default_factory=NoitaFloat)
    affect_flying:NoitaBool = field(default_factory=NoitaBool)
    ray_skip_material:NoitaInt = field(default_factory=NoitaInt)
    mTarget:NoitaCVector2 = field(default_factory=lambda: NoitaCVector2(NoitaFloat))
    mPrevTarget:NoitaCVector2 = field(default_factory=lambda: NoitaCVector2(NoitaFloat))
    mPrevCenterPosition:NoitaCVector2 = field(default_factory=lambda: NoitaCVector2(NoitaFloat))
    mState:NoitaInt = field(default_factory=NoitaInt)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.ground_attachment_min_spread.Read(s)
        self.ground_attachment_max_tries.Read(s)
        self.ground_attachment_max_angle.Read(s)
        self.ground_attachment_ray_length_coeff.Read(s)
        self.leg_velocity_coeff.Read(s)
        self.affect_flying.Read(s)
        self.ray_skip_material.Read(s)
        self.mTarget.Read(s)
        self.mPrevTarget.Read(s)
        self.mPrevCenterPosition.Read(s)
        self.mState.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.ground_attachment_min_spread.Write(s)
        self.ground_attachment_max_tries.Write(s)
        self.ground_attachment_max_angle.Write(s)
        self.ground_attachment_ray_length_coeff.Write(s)
        self.leg_velocity_coeff.Write(s)
        self.affect_flying.Write(s)
        self.ray_skip_material.Write(s)
        self.mTarget.Write(s)
        self.mPrevTarget.Write(s)
        self.mPrevCenterPosition.Write(s)
        self.mState.Write(s)

